### PR TITLE
8302043: [AIX] Safefetch fails for bad_addressN and bad_address32

### PR DESCRIPTION
**Pull request by varada1110**
On AIX, reading from lower range addresses like 5120(5K) does not cause a segmentation fault, though writing to them does. Hence the safefetch tests don't run as expected on AIX. This is solved by setting bad_addressN and bad_address32 to -1 for AIX. 
These tests under NMTGtests.java also passes :
gtest/NMTGtests.java#nmt-detail
gtest/NMTGtests.java#nmt-summary
gtest/NMTGtests.java#nmt-off

Reported issue: [JDK-8302043](https://bugs.openjdk.org/browse/JDK-8302043)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8302043](https://bugs.openjdk.org/browse/JDK-8302043): [AIX] Safefetch fails for bad_addressN and bad_address32

### Reviewers
 * [Thomas Stuefe](https://openjdk.org/census#stuefe) (@tstuefe - **Reviewer**)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12468/head:pull/12468` \
`$ git checkout pull/12468`

Update a local copy of the PR: \
`$ git checkout pull/12468` \
`$ git pull https://git.openjdk.org/jdk pull/12468/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12468`

View PR using the GUI difftool: \
`$ git pr show -t 12468`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12468.diff">https://git.openjdk.org/jdk/pull/12468.diff</a>

</details>

### CODE REVIEW

**Review by tstuefe — CHANGES_REQUESTED** [review]
Hi @varada1110,

thank you. But could you please adjust VMError::segfault_address instead? It already adjusts for AIX, but apparently that's not enough:

```
static constexpr intptr_t segfault_address = (1 * K) AIX_ONLY(+ (4 * K));
```

Note that this address is checked in java (see test/hotspot/jtreg/runtime/ErrorHandling/TestSigInfoInHsErrFile.java), which you should adjust too.

This begs the question, if VMError::segfault_address is wrong today for AIX, TestSigInfoInHsErrFile.java must have failed, too, right?

**Comment by varada1110** [comment]
> Hi @varada1110,
> 
> thank you. But could you please adjust VMError::segfault_address instead? It already adjusts for AIX, but apparently, that's not enough:
> 
> ```
> static constexpr intptr_t segfault_address = (1 * K) AIX_ONLY(+ (4 * K));
> ```
> 
> Note that this address is checked in jtreg (see test/hotspot/jtreg/runtime/ErrorHandling/TestSigInfoInHsErrFile.java), which you should adjust too (we expect the fault address to show up as fault address in the hs-err file).
> 
> This begs the question, if VMError::segfault_address is wrong today for AIX, TestSigInfoInHsErrFile.java must have failed, too, right?
> 
> Thanks, Thomas

Thanks @tstuefe for pointing this out. Updating the segfault_address is the better fix to this issue. The test/hotspot/jtreg/runtime/ErrorHandling/TestSigInfoInHsErrFile.java is also checking the segfault_address .I will update it.

File: src/hotspot/share/utilities/vmError.hpp [hunk 1]
```diff
@@ -188,8 +188,11 @@ class VMError : public AllStatic {
   DEBUG_ONLY(static void controlled_crash(int how);)
 
   // Non-null address guaranteed to generate a SEGV mapping error on read, for test purposes.
-  static constexpr intptr_t segfault_address = (1 * K) AIX_ONLY(+ (4 * K));
-
+#ifdef AIX
+  static constexpr intptr_t segfault_address = -1;
+#else
+  static constexpr intptr_t segfault_address = (1*K);
+#endif //AIX
```
Comment (tstuefe) [inline src/hotspot/share/utilities/vmError.hpp:195]: Nit, could you please use 
```
static constexpr intptr_t segfault_address = AIX_ONLY(-1) NOT_AIX(1*K);
```
instead? Its a bit tighter coding.

Thanks!

Comment (varada1110) [reply src/hotspot/share/utilities/vmError.hpp:195]: Okay. Thank you

**Review by dholmes-ora — APPROVED** [review]
No impact on non-AIX so trivially approved under the assumption the tests now pass on AIX.

Thanks.

**Comment by tstuefe** [comment]
Test fix still missing... see 

https://github.com/openjdk/jdk/blob/bf17936d1a3aa86e4b581605ed1a15c8e66c1c3b/test/hotspot/jtreg/runtime/ErrorHandling/TestSigInfoInHsErrFile.java#L71

**Comment by varada1110** [comment]
> Test fix still missing... see
> 
> https://github.com/openjdk/jdk/blob/bf17936d1a3aa86e4b581605ed1a15c8e66c1c3b/test/hotspot/jtreg/runtime/ErrorHandling/TestSigInfoInHsErrFile.java#L71

Yes I have made those changes but /test/hotspot/jtreg/runtime/ErrorHandling/TestSigInfoInHsErrFile.java is failing before  checking VMError::_segfault_address.
TestSigInfoInHsErrFile.java is failing with a RuntimeException: hs-err file incomplete (first missing pattern: # .*VMError::controlled_crash.*). 

Thanks

**Comment by tstuefe** [comment]
> > Test fix still missing... see
> > https://github.com/openjdk/jdk/blob/bf17936d1a3aa86e4b581605ed1a15c8e66c1c3b/test/hotspot/jtreg/runtime/ErrorHandling/TestSigInfoInHsErrFile.java#L71
> 
> Yes I have made those changes and checked but /test/hotspot/jtreg/runtime/ErrorHandling/TestSigInfoInHsErrFile.java is failing before checking VMError::_segfault_address. TestSigInfoInHsErrFile.java is failing with a RuntimeException: hs-err file incomplete (first missing pattern: # ._VMError::controlled_crash._).
> 
> Thanks

That is a different issue; probably related to `VMError::controlled_crash()` being inlined by xlc, so it does not show up in the callstack. There have been fixes for this around, I think.

Okay, I approve this now, but if you fix the VMError::controlled_crash() problem in TestSigInfoInHsErrFile.java, you will run right into the problem that "0x0*1400" does not match. Then you have to change this to (I guess) 0xFFFFFFFFFFFFFFFF (your -1 you pass as invalid address).

**Review by tstuefe — APPROVED** [review]
Okay, thanks.

**Comment by varada1110** [comment]
> > > Test fix still missing... see
> > > https://github.com/openjdk/jdk/blob/bf17936d1a3aa86e4b581605ed1a15c8e66c1c3b/test/hotspot/jtreg/runtime/ErrorHandling/TestSigInfoInHsErrFile.java#L71
> > 
> > 
> > Yes I have made those changes and checked but /test/hotspot/jtreg/runtime/ErrorHandling/TestSigInfoInHsErrFile.java is failing before checking VMError::_segfault_address. TestSigInfoInHsErrFile.java is failing with a RuntimeException: hs-err file incomplete (first missing pattern: # ._VMError::controlled_crash._).
> > Thanks
> 
> That is a different issue; probably related to `VMError::controlled_crash()` being inlined by xlc, so it does not show up in the callstack. There have been fixes for this around, I think.
> 
> Okay, I approve this now, but if you fix the VMError::controlled_crash() problem in TestSigInfoInHsErrFile.java, you will run right into the problem that "0x0*1400" does not match. Then you have to change this to (I guess) 0xFFFFFFFFFFFFFFFF (your -1 you pass as invalid address).

Okay. Thank you

**Pull request closed without merging**